plugins/tar/receive-app

```diff
@@ -11,7 +12,10 @@ tar_receive_app() {
   if [[ ! -f "$DOKKU_ROOT/$APP/src.tar" ]]; then
     true
   else
-    dokku tar:build "$APP"
+    acquire_app_deploy_lock "$APP"
+    # shellcheck disable=SC2086
+    tar_build "$@"
+    release_app_deploy_lock "$APP"
```

Comment:
That implementation only worked because we used `flock -o "$APP_BUILD_LOCK" dokku tar:build-locked "$@"`. That is not possible when calling functions.